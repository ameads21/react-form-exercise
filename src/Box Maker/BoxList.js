import React, { useState } from "react";
import { v4 as uuid } from "uuid";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
const BoxList = () => {
  const [boxes, setBoxes] = useState([]);
  const addBox = (newBox) => {
    setBoxes((boxes) => [...boxes, { ...newBox, id: uuid() }]);
  };
  const remove = (id) => {
    setBoxes((boxes) => boxes.filter((box) => box.id !== id));
  };

  const boxComponents = boxes.map((box) => (
    <Box
      id={box.id}
      color={box.color}
      width={box.width}
      height={box.height}
      key={box.id}
      handleRemove={remove}
    />
  ));

  return (
    <div>
      <h1>Color Box Maker</h1>
      <NewBoxForm addBox={addBox} />
      {boxComponents}
    </div>
  );
};

export default BoxList;
