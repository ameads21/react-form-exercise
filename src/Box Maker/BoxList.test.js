import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function () {
  render(<BoxList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new box", function () {
  const { queryByText, getByLabelText } = render(<BoxList />);
  const input = getByLabelText("Color");
  const btn = queryByText("Create Box");
  expect(queryByText("Remove The Box!")).not.toBeInTheDocument();
  fireEvent.change(input, { target: { value: "purple" } });
  fireEvent.click(btn);
  expect(queryByText("Remove The Box!")).toBeInTheDocument();
});
