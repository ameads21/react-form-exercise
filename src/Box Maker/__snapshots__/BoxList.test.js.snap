// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Color Box Maker
    </h1>
    <form>
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        placeholder="Color Name"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        min="0"
        name="width"
        type="number"
        value="0"
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        min="0"
        name="height"
        type="number"
        value="0"
      />
      <button>
        Create Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
